Use OpenBSD numbering for shared library.

Provides auxiliary vector support for OpenBSD
69ad604ff8ba35860cfb22be3ed4e763ea07ba2c

Index: src/build-data/os/openbsd.txt
--- src/build-data/os/openbsd.txt.orig
+++ src/build-data/os/openbsd.txt
@@ -1,7 +1,7 @@
 
 soname_pattern_base  "lib{libname}.so"
-soname_pattern_abi   "lib{libname}.so.{abi_rev}.{version_minor}"
-soname_pattern_patch "lib{libname}.so.{abi_rev}.{version_minor}"
+soname_pattern_abi   "lib{libname}.so.${LIBbotan-2_VERSION}"
+soname_pattern_patch "lib{libname}.so.${LIBbotan-2_VERSION}"
 
 shared_lib_symlinks no
 
@@ -13,6 +13,7 @@ posix_mlock
 clock_gettime
 dev_random
 arc4random
+elf_aux_info
 getentropy
 explicit_bzero
 pledge
